src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ConcurrencyLimiter.cs

```diff
@@ -59,7 +71,11 @@ protected override RateLimitLease AcquireCore(int permitCount)
             // Return SuccessfulLease or FailedLease to indicate limiter state
             if (permitCount == 0)
             {
-                return _permitCount > 0 ? SuccessfulLease : FailedLease;
+                if (_permitCount > 0)
+                {
+                    return SuccessfulLease;
+                }
+                return FailedLease;
```

Comment:
Is the theory here that when the `permitCount` is 0, the caller is just checking, so this isn't a "real" failed lease? I'm not sure I agree with this because I don't think this is the most straightforward way to interpret `TotalFailedLeases`.

I think we should probably count these. But if we don't, we should at least note this behavior in `TotalSuccessfulLeases` and `TotalFailedLeases`.